k2/csrc/intersect.cu

```diff
@@ -90,950 +69,840 @@ using namespace intersect_internal;  // NOLINT
 // Caution: this is really a .cu file.  It contains mixed host and device code.
 
 /*
-   Pruned intersection (a.k.a. composition) that corresponds to decoding for
-   speech recognition-type tasks.  Can use either different decoding graphs (one
-   per acoustic sequence) or a shared graph
+   Intersection (a.k.a. composition) that corresponds to decoding for
+   speech recognition-type tasks.  This version does only forward-backward
+   pruning in the backward pass; the forward pass does no pruning.
+
+   Can use either different decoding graphs (one per acoustic sequence) or a
+   shared graph.
+
+   How to use this object:
+       Construct it
+       Call Intersect()
+       Call FormatOutput()
 */
 class MultiGraphDenseIntersect {
  public:
   /**
-     Pruned intersection (a.k.a. composition) that corresponds to decoding for
+     Intersection (a.k.a. composition) that corresponds to decoding for
      speech recognition-type tasks
 
        @param [in] a_fsas  The decoding graphs, one per sequence.  E.g. might
                            just be a linear sequence of phones, or might be
                            something more complicated.  Must have either the
-                           same Dim0() as b_fsas, or Size0()==1 in which
+                           same Dim0() as b_fsas, or Dim0() == 1 in which
                            case the graph is shared.
        @param [in] b_fsas  The neural-net output, with each frame containing the
                            log-likes of each phone.  A series of sequences of
-                           (in general) different length.
-       @param [in] beam    "Default" decoding beam.  The actual beam is dynamic
-                            and also depends on max_active and min_active.
-       @param [in] min_active  Minimum number of FSA states that are allowed to
-                           be active on any given frame for any given
-                           intersection/composition task. This is advisory,
-                           in that it will try not to have fewer than this
-                           number active.
-       @param [in] max_active  Maximum number of FSA states that are allowed to
-                           be active on any given frame for any given
-                           intersection/composition task. This is advisory,
-                           in that it will try not to exceed that but may not
-                           always succeed.
+                           (in general) different length.  MUST BE SORTED BY
```

Comment:
> MUST BE SORTED BY DECREASING LENGTH, or it is an error.

We need to change the Python code of constructing DenseFsaVec.
The current implementation does not satisfy this constraint.

Reply:
You're right, needs change.  Should be easy though.

Reply:
Should the sort be performed by the user or k2?
If it is done by the user, do we need to check that the input supervision segments are sorted?